Analyzers/AnchorPointAnalyzer.Extractor.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace ProblematicBookmark.Analyzers
+{
+    using System.Text.RegularExpressions;
+
+    internal partial class AnchorPointAnalyzer
+    {
+        private static readonly string Pattern = @"\]\((.*?)\)";
+
+        private async Task Extract(string content)
+        {
+            if (string.IsNullOrEmpty(content))
+            {
+                await Task.CompletedTask;
+            }
+
+            await ExtractAnchorPointByRegex(content);
+        }
+
+        private async Task ExtractAnchorPointByRegex(string content)
+        {
+            var match = Regex.Match(content, Pattern);
+            while (match.Success)
+            {
+                var anchorPoint = await GetAnchorPoint(match.Groups[1].Value);
+                if (!string.IsNullOrEmpty(anchorPoint))
+                {
+                    Add(KeyValuePair.Create(anchorPoint, match.Groups[1].Value));
+                }
+
+                match = match.NextMatch();
+            }
+        }
+
+        private async static Task<string> GetAnchorPoint(string str)
+        {
+
+            if (string.IsNullOrEmpty(str) || !str.StartsWith("/"))
+            {
+                return string.Empty;
+            }
+
+            if (str.StartsWith("https://docs.microsoft.com"))

```

Comment:
if not found #, we need to return string.Empty

Reply:
it has been handled.

var startIndex = str.IndexOf("#");
            if (startIndex == -1 || startIndex == str.Length)
            {
                return string.Empty;
            }

Reply:
we need return string.Empty before check str.StartsWith("https://docs.microsoft.com"),
if if (str.StartsWith("https://docs.microsoft.com")) but the str does not have #, we don't concern such url

Reply:
Updated